File: University/lib/Dto/Training/JurusanDto.dart
class JurusanDto {
  String kode_fakultas = "";
  String kode_jurusan = "";
  String nama_jurusan = "";
  double record_status = 0;

  bool isSelected = false;
  double PageNumber = 0;
  double PageSize = 0;
  double TotalPage = 0;
  double TotalRecord = 0;

  JurusanDto({
    this.kode_fakultas = "",
    this.kode_jurusan = "",
    this.nama_jurusan = "",
    this.record_status = 0,
    this.isSelected = false,
    this.PageNumber = 0,
    this.PageSize = 0,
    this.TotalPage = 0,
    this.TotalRecord = 0,
  });

  factory JurusanDto.fromJson(Map<String, dynamic> json) {
    return JurusanDto(
      kode_fakultas: json["kode_fakultas"] ?? "",
      kode_jurusan: json["kode_jurusan"] ?? "",
      nama_jurusan: json["nama_jurusan"] ?? "",
      record_status: json["record_status"] ?? 0,
      isSelected: json["isSelected"] ?? false,
      PageNumber: json["PageNumber"] ?? 0,
      PageSize: json["PageSize"] ?? 0,
      TotalPage: json["TotalPage"] ?? 0,
      TotalRecord: json["TotalRecord"] ?? 0,
    );
  }

  Map toMap() {
    var map = <String, dynamic>{};
    map["kode_fakultas"] = kode_fakultas ?? "";
    map["kode_jurusan"] = kode_jurusan ?? "";
    map["nama_jurusan"] = nama_jurusan ?? "";
    map["record_status"] = record_status ?? 0;
    map["isSelected"] = isSelected;
    map["PageNumber"] = PageNumber;
    map["PageSize"] = PageSize;
    map["TotalPage"] = TotalPage;
    map["TotalRecord"] = TotalRecord;
    return map;
  }
}

File: University.Service/University.Dto/Training/JurusanDto.cs
using System;
using System.Data;
using System.Collections.Generic;
using System.Runtime.Serialization;

using University.Dto.Base;

namespace University.Dto.Training
{
    [DataContract]
    public class JurusanDto
    {
        #region General Property
        [DataMember] public string kode_fakultas { get; set; }
        [DataMember] public string kode_jurusan { get; set; }
        [DataMember] public string nama_jurusan { get; set; }
        [DataMember] public decimal record_status { get; set; }

        #endregion

        #region Additional Property

        [DataMember]
        public int PageNumber { get; set; }

        [DataMember]
        public int PageSize { get; set; }

        [DataMember]
        public int TotalPage { get; set; }

        [DataMember]
        public int TotalRecord { get; set; }

        [DataMember]
        public bool IsSelected { get; set; }

        [DataMember]
        public string SqlFilter { get; set; }

        [DataMember]
        public string SqlSort { get; set; }
        #endregion
    }

    public class JurusanMappingDto : Mapper<JurusanDto>
    {
        protected override JurusanDto PopulateItem(IDataRecord dr)
        {
            JurusanDto dto = new JurusanDto();
            for (int i = 0; i < dr.FieldCount; i++)
            {
                MapProperty(dto, dr.GetName(i), dr[i]);
            }
            return dto;
        }
    }
}
